src/HDF5.jl

```diff
@@ -426,10 +426,8 @@ hash(x::HDF5ReferenceObj, h::UInt) = hash(x.r, h)
 # Compound types
 # These are "raw" and not mapped to any Julia type
```

Comment:
I guess we need to update the comment now that the data is mapped to Julia types.